serializable.ts

```diff
@@ -148,16 +148,7 @@ export function toPojo(
       toJSONStrategy = toJSONRecursive;
     }
 
-    if (Array.isArray(value)) {
-      record[serializedKey] = value.map((item: JSONValue) => {
-        if (item instanceof Serializable) {
-          return toPojo(item);
-        }
-        return toJSONStrategy(item);
-      });
-    } else if (value !== undefined) {
-      record[serializedKey] = toJSONStrategy(value);
-    }
+    record[serializedKey] = toJSONStrategy(value);
```

Comment:
I think there's an important change to the default behaviour for serialization here. Previously, if you had an array property and were using the default strategy:

```
@SerializeProperty()
private someArray: SomeClass[];
```
... we'd automatically call toPojo() on each element in the array.

Now the default is that we'll call toJSONStrategy(), which just returns the original value. So serialization that would have worked properly before by default, no longer will.

Granted, you always had to give us more information for _deserialization_ to work, using `toSerializable()` in a `fromJSONStrategy`. But previously you could get away without a `toJSONStrategy` and it would do the right thing, kind of by accident, if you only cared about that direction. So that might be worth updating in the examples.

For our own use, we use a shortcut decorator `@DeserializeAs()`:

```ts
export function DeserializeAs<T>(type: SerializableConstructor<T>, propertyName?: string): PropertyDecorator {
    const objectSerializer: SerializePropertyArgument = {
        fromJSONStrategy: toSerializable(type),
    };
    if (propertyName) {
        objectSerializer.serializedKey = propertyName;
    }

    return SerializeProperty(objectSerializer);
}
```
... we would just need to change it to specify a toJsonStrategy of fromSerializable(), so it's not a big change.

Reply:
Yes 💯 

This is a breaking change and will not be released until v2 is ready. There are a few other features for us to take care of and I only recently and started to set up and plan v2. I am going to add `new Example().clone(arg: Partial<Example>)` or something similar, and we are going to start supporting common decorators, although I'd like to discuss how that will look before starting.

Reply:
Another breaking change is with `.fromJSON()` it only takes a `JSONValue` now, IDK if this is the best approach but I currently believe it is better than using `Object` to support angular and rather have integrators use typing with angular.

Again that is also up for discussion.

Reply:
oh damn! I just noticed I am bringing in a bug by allowing `undefined` to be processed by strategies. 

Thanks, @ms1111